src/SFA.DAS.LevyTransferMatching.Web/Controllers/PledgesController.cs

```diff
@@ -232,10 +233,51 @@ private void AddLocationErrorsToModelState(Dictionary<int, string> errors)
         public async Task<IActionResult> Applications(ApplicationsRequest request)
         {
             var response = await _orchestrator.GetApplications(request);
-
             return View(response);
         }
 
+        [Authorize(Policy = PolicyNames.ManageAccount)]
+        [HttpPost]
+        [Route("{encodedPledgeId}/applications")]
+        public IActionResult Applications(ApplicationsPostRequest request)
+        {
+            return RedirectToAction(nameof(RejectApplications), new { request.EncodedAccountId, request.EncodedPledgeId, request.ApplicationsToReject });
+        }
+
+        
+        [Authorize]
+        [Route("{encodedPledgeId}/applications/reject-applications")]
+        public async Task<IActionResult> RejectApplications(ApplicationsPostRequest request)
+        {
+            var getRejectApplicationsViewModel = await _orchestrator.GetRejectApplicationsViewModel(request);
+            return View(getRejectApplicationsViewModel);
+        }
+
+        [Authorize]
+        [Route("{encodedPledgeId}/applications/reject-applications")]
+        [HttpPost]
+        public async Task<IActionResult> RejectApplications(RejectApplicationsPostRequest request)
+        {
+            string numberOfApplicationsRejectedBanner = string.Empty;
+
+            if (request.RejectConfirm)
+            {
+                await _orchestrator.RejectApplications(request);
+
+                if (request.ApplicationsToReject?.Count == 1)
```

Comment:
Could we think of somewhere to put this, or refactor it down, as it's bloating the action body?

Reply:
refactored the code into a private method  SetRejectedApplicationsBanner(count);